boot.py

```diff
@@ -86,10 +84,86 @@
 from .plugin.tooling import LspOnDoubleClickCommand
 from .plugin.tooling import LspParseVscodePackageJson
 from .plugin.tooling import LspTroubleshootServerCommand
-from typing import Any, Dict, List, Optional, Type
+from typing import Any
 
+__all__ = (
+    "DocumentSyncListener",
+    "Listener",
+    "LspApplyDocumentEditCommand",
+    "LspApplyWorkspaceEditCommand",
+    "LspCallHierarchyCommand",
+    "LspClearLogPanelCommand",
+    "LspClearPanelCommand",
+    "LspCodeActionsCommand",
+    "LspCodeLensCommand",
+    "LspCollapseTreeItemCommand",
+    "LspColorPresentationCommand",
+    "LspCommitCompletionWithOppositeInsertMode",
+    "LspCopyToClipboardFromBase64Command",
+    "LspDisableLanguageServerGloballyCommand",
+    "LspDisableLanguageServerInProjectCommand",
+    "LspDocumentSymbolsCommand",
+    "LspDumpBufferCapabilities",
+    "LspDumpWindowConfigs",
+    "LspEnableLanguageServerGloballyCommand",
+    "LspEnableLanguageServerInProjectCommand",
+    "LspExecuteCommand",
+    "LspExpandSelectionCommand",
+    "LspExpandTreeItemCommand",
+    "LspFoldAllCommand",
+    "LspFoldCommand",
+    "LspFormatCommand",
+    "LspFormatDocumentCommand",
+    "LspFormatDocumentRangeCommand",
+    "LspGotoDiagnosticCommand",
+    "LspHideRenameButtonsCommand",
+    "LspHierarchyToggleCommand",
+    "LspHoverCommand",
+    "LspInlayHintClickCommand",
+    "LspNextDiagnosticCommand",
+    "LspOnDoubleClickCommand",
+    "LspOpenLinkCommand",
+    "LspOpenLocationCommand",
+    "LspParseVscodePackageJson",
+    "LspPrevDiagnosticCommand",
+    "LspRefactorCommand",
+    "LspResolveDocsCommand",
+    "LspRestartServerCommand",
+    "LspRunTextCommandHelperCommand",
+    "LspSaveAllCommand",
+    "LspSaveCommand",
+    "LspSelectCompletionCommand",
+    "LspSelectionAddCommand",
+    "LspSelectionClearCommand",
+    "LspSelectionSetCommand",
+    "LspShowDiagnosticsPanelCommand",
+    "LspShowScopeNameCommand",
+    "LspSignatureHelpNavigateCommand",
+    "LspSignatureHelpShowCommand",
+    "LspSourceActionCommand",
+    "LspSymbolDeclarationCommand",
+    "LspSymbolDefinitionCommand",
+    "LspSymbolImplementationCommand",
+    "LspSymbolReferencesCommand",
+    "LspSymbolRenameCommand",
+    "LspSymbolTypeDefinitionCommand",
+    "LspToggleCodeLensesCommand",
+    "LspToggleHoverPopupsCommand",
+    "LspToggleInlayHintsCommand",
+    "LspToggleLogPanelLinesLimitCommand",
+    "LspToggleServerPanelCommand",
+    "LspTroubleshootServerCommand",
+    "LspTypeHierarchyCommand",
+    "LspUpdateLogPanelCommand",
+    "LspUpdatePanelCommand",
+    "LspWorkspaceSymbolsCommand",
+    "TextChangeListener",
+    "plugin_loaded",
+    "plugin_unloaded",
+)
```

Comment:
the `__all__` section looks  a bit too boilerplatey.

I do not think that we have any notable performance improvements.
mostly the `__all__` only helps when `import *` is used, it still doesn't restrict if someone imports stuff like `from LSP.boot import opening_files`

So I am a bit skeptical if we should add it.

Python code is usually readable, 
this feels like it is unnecessary and boilerplate.

Reply:
Are there any pros to using `__all__` in practice?

Reply:
I am fine either way, if you prefer noqa.

Reply:
> Are there any pros to using `__all__` in practice?

Make the interface clear? Probably we dont' care though.

Reply:
I would prefer `__all__` to be honest but I'm afraid this will be a breaking change. It would have to be verified by someone who has ALL the LSP packages installed/checked out and see if there are any import errors.

Reply:
I can revert that change so that it won't be a not-so-meaningful blocker to this PR.

Reply:
No packages complain.

<details>

```
reloading plugin LSP-prisma.plugin
reloading python 3.3 plugin ColorPicker.sublimecp
reloading python 3.3 plugin lsp_maintainers.main
reloading python 3.3 plugin UnitTesting33.plugin
reloading plugin Package Control.plugin
reloading plugin Sass.plugin
reloading plugin AutomaticPackageReloader.package_reloader
reloading plugin Debugger.start
reloading plugin LSP.boot
reloading plugin LSP-angular.plugin
reloading plugin LSP-astro.plugin
reloading plugin LSP-aurelia.plugin
reloading plugin LSP-basedpyright.commands
reloading plugin LSP-basedpyright.plugin
reloading plugin LSP-bash.plugin
reloading plugin LSP-Bicep.plugin
reloading plugin LSP-biome.plugin
reloading plugin LSP-clangd.plugin
reloading plugin LSP-cmake.plugin
reloading plugin LSP-copilot.boot
reloading plugin LSP-cspell.plugin
reloading plugin LSP-cspell.types
reloading plugin LSP-css.data_types
reloading plugin LSP-css.plugin
reloading plugin LSP-Dart.plugin
reloading plugin LSP-Deno.plugin
reloading plugin LSP-dockerfile.plugin
reloading plugin LSP-elixir.plugin
reloading plugin LSP-elixir.server_zip_resource
reloading plugin LSP-elm.plugin
reloading plugin LSP-elm.types
reloading plugin LSP-eslint.plugin
reloading plugin LSP-file-watcher-chokidar.watcher
reloading plugin LSP-flow.plugin
reloading plugin LSP-gnols.plugin
reloading plugin LSP-gopls.boot
reloading plugin LSP-Grammarly.plugin
reloading plugin LSP-Grammarly.plugin_commands
reloading plugin LSP-graphql.plugin
reloading plugin LSP-html.data_types
reloading plugin LSP-html.plugin
reloading plugin LSP-intelephense.plugin
reloading plugin LSP-jdtls.boot
reloading plugin LSP-json.commands
reloading plugin LSP-json.plugin
reloading plugin LSP-json.schema_store
reloading plugin LSP-julia.plugin
reloading plugin LSP-kotlin.plugin
reloading plugin LSP-lemminx.plugin
reloading plugin LSP-leo.plugin
reloading plugin LSP-ltex-ls.plugin
reloading plugin LSP-lua.plugin
reloading plugin LSP-marksman.plugin
reloading plugin LSP-metals.plugin
reloading plugin LSP-nimlangserver.plugin
reloading plugin LSP-OmniSharp.plugin
reloading plugin LSP-performance-test.__init__
reloading plugin LSP-performance-test.test_completions
reloading plugin LSP-PowerShellEditorServices.plugin
reloading plugin LSP-pylsp.plugin
reloading plugin LSP-pyright.commands
reloading plugin LSP-pyright.plugin
reloading plugin LSP-R.plugin
reloading plugin LSP-rome.plugin
reloading plugin LSP-ruff.plugin
reloading plugin LSP-rust-analyzer.plugin
reloading plugin LSP-rust-analyzer.plugin_commands
reloading plugin LSP-SonarLint.plugin
reloading plugin LSP-SourceKit.plugin
reloading plugin LSP-stylelint.plugin
reloading plugin LSP-svelte.plugin
reloading plugin LSP-tailwindcss.plugin
reloading plugin LSP-terraform.plugin
reloading plugin LSP-TexLab.const
reloading plugin LSP-TexLab.lsp_texlab_auto_complete
reloading plugin LSP-TexLab.lsp_texlab_build_system
reloading plugin LSP-TexLab.plugin
reloading plugin LSP-TexLab.server
reloading plugin LSP-TexLab.tarball
reloading plugin LSP-typescript.commands
reloading plugin LSP-typescript.plugin
reloading plugin LSP-typescript.plugin_types
reloading plugin LSP-vale-ls.plugin
reloading plugin LSP-VHDL-ls.plugin
reloading plugin LSP-volar.plugin
reloading plugin LSP-volar.volar_virtual_file_hierarchy
reloading plugin LSP-volar.volar_write_virtual_files
reloading plugin LSP-vue.plugin
reloading plugin LSP-yaml.plugin
```

</details>

Reply:
Well I think that most people here said that the `__all__` option is sane
so I think the conclusion is that `__all__` stays.

The packages load without errors, 
while this can be a breaking change, 
it actually will not break any package.